parastell/invessel_build.py

```diff
@@ -49,134 +53,99 @@ def orient_spline_surfaces(volume_id):
-            value is greater than the number of entries in 'poloidal_angles'.
-        scale (double): a scaling factor between the units of VMEC and [cm]
+            (defaults to 61). Points are set at poloidal angles interpolated
+            between those specified in 'poloidal_angles' if this value is
+            greater than the number of entries in 'poloidal_angles'.
+        scale (float): a scaling factor between the units of VMEC and [cm]
             (defaults to m2cm = 100).
-        plasma_mat_tag (str): alternate DAGMC material tag to use for plasma.
-            If none is supplied, 'plasma' will be used (defaults to None).
-        sol_mat_tag (str): alternate DAGMC material tag to use for scrape-off
-            layer. If none is supplied, 'sol' will be used (defaults to None).
-        logger (object): logger object (defaults to None). If no logger is
-            supplied, a default logger will be instantiated.
     """
 
     def __init__(
-            self,
-            vmec,
-            toroidal_angles,
-            poloidal_angles,
-            radial_build,
-            wall_s,
-            repeat=0,
-            num_ribs=61,
-            num_rib_pts=61,
-            scale=m2cm,
-            plasma_mat_tag=None,
-            sol_mat_tag=None,
-            logger=None
+        self,
+        vmec_obj,
+        radial_build,
+        logger=None,
+        **kwargs
     ):
-        self.vmec = vmec
-        self.toroidal_angles = toroidal_angles
-        self.poloidal_angles = poloidal_angles
-        self.radial_build = radial_build
-        self.wall_s = wall_s
-        self.repeat = repeat
-        self.num_ribs = num_ribs
-        self.num_rib_pts = num_rib_pts
-        self.scale = scale
-        self.plasma_mat_tag = plasma_mat_tag
-        self.sol_mat_tag = sol_mat_tag
 
         self.logger = logger
-        if self.logger == None or not self.logger.hasHandlers():
-            self.logger = log.init()
+        self.vmec_obj = vmec_obj
+        self.radial_build = radial_build
+        
+        self.repeat = 0
+        self.num_ribs = 61
+        self.num_rib_pts = 67
+        self.scale = m2cm
+
+        allowed_kwargs = ['repeat', 'num_ribs', 'num_rib_pts', 'scale']
+        set_kwarg_attrs(
+            self,
+            kwargs,
+            allowed_kwargs,
+            self._logger
+        )
 
         self.Surfaces = {}
         self.Components = {}
 
-        if (self.repeat + 1) * self.toroidal_angles[-1] > 360.0:
+    @property
+    def vmec_obj(self):
+        return self._vmec_obj
+    
+    @vmec_obj.setter
+    def vmec_obj(self, vmec_object):
+        self._vmec_obj = vmec_object
+
+    @property
+    def logger(self):
+        return self._logger
+    
+    @logger.setter
+    def logger(self, logger_object):
+        self._logger = logger_object
+        if self._logger == None or not self._logger.hasHandlers():
+            self._logger = log.init()
```

Comment:
Since we are doing this in every class, what if we add `log.check_init(logger_object)`
```
# in log.py
def check_init(logger_object):
    if logger_object != None and not logger_object.hasHandlers():
        return logger_object
    else
       return log.init()
```

And this method then is just one line:
```
self._logger = log.check_init(logger_object)
```